### DISCUSSION TITLE
Error: uninitialized constant Kernel::GitHub

### DISCUSSION THREAD

**Discussion by chuckwondo** [Homebrew/discussions#5674] [Everyday usage] [answered]
### Output of `brew config`

```shell
Error: uninitialized constant Kernel::GitHub
Warning: Removed Sorbet lines from backtrace!
Rerun with `--verbose` to see the original backtrace
/usr/local/Homebrew/Library/Homebrew/extend/kernel.rb:67:in `opoo'
/usr/local/Homebrew/Library/Homebrew/extend/kernel.rb:183:in `odeprecated'
/usr/local/Homebrew/Library/Homebrew/env_config.rb:522:in `block (2 levels) in <module:EnvConfig>'
/usr/local/Homebrew/Library/Homebrew/cli/parser.rb:733:in `public_send'
/usr/local/Homebrew/Library/Homebrew/cli/parser.rb:733:in `value_for_env'
/usr/local/Homebrew/Library/Homebrew/cli/parser.rb:228:in `switch'
/usr/local/Homebrew/Library/Homebrew/cli/parser.rb:192:in `block in initialize'
/usr/local/Homebrew/Library/Homebrew/cli/parser.rb:191:in `each'
/usr/local/Homebrew/Library/Homebrew/cli/parser.rb:191:in `initialize'
/usr/local/Homebrew/Library/Homebrew/brew.rb:51:in `new'
/usr/local/Homebrew/Library/Homebrew/brew.rb:51:in `<main>'
```

### Output of `brew doctor`

```shell
Error: uninitialized constant Kernel::GitHub
Warning: Removed Sorbet lines from backtrace!
Rerun with `--verbose` to see the original backtrace
/usr/local/Homebrew/Library/Homebrew/extend/kernel.rb:67:in `opoo'
/usr/local/Homebrew/Library/Homebrew/extend/kernel.rb:183:in `odeprecated'
/usr/local/Homebrew/Library/Homebrew/env_config.rb:522:in `block (2 levels) in <module:EnvConfig>'
/usr/local/Homebrew/Library/Homebrew/cli/parser.rb:733:in `public_send'
/usr/local/Homebrew/Library/Homebrew/cli/parser.rb:733:in `value_for_env'
/usr/local/Homebrew/Library/Homebrew/cli/parser.rb:228:in `switch'
/usr/local/Homebrew/Library/Homebrew/cli/parser.rb:192:in `block in initialize'
/usr/local/Homebrew/Library/Homebrew/cli/parser.rb:191:in `each'
/usr/local/Homebrew/Library/Homebrew/cli/parser.rb:191:in `initialize'
/usr/local/Homebrew/Library/Homebrew/brew.rb:51:in `new'
/usr/local/Homebrew/Library/Homebrew/brew.rb:51:in `<main>'
```

### Description of issue

I ran `brew bundle --global` to update my installed formulae/casks and after brew updated itself, I started getting `Error: uninitialized constant Kernel::GitHub` for every `brew` command I ran.

I pinpointed the problem to this function: https://github.com/Homebrew/brew/blob/master/Library/Homebrew/extend/kernel.rb#L66-L72

I locally added the following lines to the top of that function in my local clone:

```
    require "utils/github/actions"
    require "utils/formatter"
```

After that, everything worked fine.

However, it seems odd that I cannot find any other reports of this error anywhere. I was going to report this as a bug, but I'm doubting myself here because I updated to macOS Sequoia yesterday, so I suspect that perhaps something got clobbered.

On the other hand, as far as I can tell, the code linked above does indeed appear to be missing the necessary `require` statements, particularly since the function that follows the linked function is nearly identical and _does_ contain those `require` statements.

**Comment by Bo98** [comment, marked as answer]
> I was going to report this as a bug, but I'm doubting myself here because I updated to macOS Sequoia yesterday, so I suspect that perhaps something got clobbered.

This is a bug, but it looks like it only happens if you have a `HOMEBREW_*` environment variable explicitly set to `0` (or `false` or `off` or similar). We started warning about this as it doesn't actually behave as people expected and has always behaved like `1`/`true`, but it seems that warning is broken.

Evidently, the number of people this applied to must be very low. Though this did only break a couple days ago and the regression has only been rolled out to those with developer mode enabled (~20%).

**Reply by chuckwondo** [reply]
Aha! Thank you! I can indeed confirm that when I set a `HOMEBREW_*` var to `false`, `off`, or `0`, I can reproduce the error. (I had `HOMEBREW_VERBOSE=false`, which was giving me this problem). When I changed the value to be empty (e.g., `HOMEBREW_VERBOSE=`) or some "truthy" value, everything worked just fine.
